ldap-api/src/test/java/ca/bc/gov/hlth/ldapapi/controller/UsersControllerTest.java

```diff
@@ -0,0 +1,196 @@
+package ca.bc.gov.hlth.ldapapi.controller;
+
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.Test;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.beans.factory.annotation.Value;
+import org.springframework.boot.test.context.SpringBootTest;
+import org.springframework.boot.test.web.client.TestRestTemplate;
+import org.springframework.boot.web.server.LocalServerPort;
+import org.springframework.http.ResponseEntity;
+
+import javax.naming.NamingEnumeration;
+import javax.naming.NamingException;
+import javax.naming.directory.*;
+import java.util.HashMap;
+import java.util.Map;
+import java.util.Properties;
+
+import static org.junit.jupiter.api.Assertions.*;
+
+@SuppressWarnings({"ConstantConditions", "rawtypes"})
+@SpringBootTest(webEnvironment = SpringBootTest.WebEnvironment.RANDOM_PORT)
+class UsersControllerTest {
+
+    @LocalServerPort
+    private int port;
+
+    private String urlUnderTest;
+
+    @Autowired
+    private TestRestTemplate restTemplate;
+
+    @Autowired
+    Properties ldapProperties;
+
+    @Value("${LDAP_API_INTEGRATION_TEST_PASSWORD}")
+    String PASSWORD;
+
+    public static final String USERNAME = "1-primehcimintegrationtest";
+
+    private static boolean runOnce = true;
+
+    @BeforeEach
+    public void init() throws InterruptedException, NamingException {
+        urlUnderTest = "http://localhost:" + port + "/users";
+        if (runOnce) {
+            SearchResult searchResult = searchUser(USERNAME);
+            unlockUserAccount(searchResult.getName() + "," + LDAP_SEARCH_BASE);
```

Comment:
...you could add a null check here (in the off chance someone deletes your user in LDAP Dev and `searchUser()` returns null).